test/test_jit.py

```diff
@@ -3566,6 +3566,70 @@ def mixtypes(x):
                 if True:
                     a = 4
 
+    def test_if_tuple_sizes(self):
+        with self.assertRaisesRegex(RuntimeError, "Type mismatch"):
+            @torch.jit.script
+            def diff_tuple_sizes(x):
+                if False:
+                    c0 = ((x, x), (x, x, x))
+                else:
+                    c0 = ((x, x, x), (x, x))
+                return c0
+
+    def test_if_different_type(self):
+        with self.assertRaisesRegex(RuntimeError, "Type mismatch: c0 is set to type Tensor "
+                                    "in the true branch and type int in the false branch:"):
+            @torch.jit.script
+            def tensor_scalar(x):
+                if False:
+                    c0 = x
+                else:
+                    c0 = 1
+                return c0
+        with self.assertRaisesRegex(RuntimeError, "Type mismatch: c0 is set to type int in "
+                                    "the true branch and type float in the false branch:"):
+            @torch.jit.script
+            def int_float(x):
+                if False:
+                    c0 = 1
+                else:
+                    c0 = 1.0
+                return c0
+
+    def test_if_list(self):
+        # testing that different length lists don't throw error
+        @torch.jit.script
+        def test_list(x):
+            if True:
+                c = [x, x]
+            else:
+                c = [x, x, x]
+            return torch.cat(c)
+
+        b = torch.zeros(2, 4)
+        # unrelated bug
```

Comment:
Can you say sth more about the bug? I don't know what's the problem here

Reply:
This is the list propagation bug that I have a PR out for.